internal/controller/pgupgrade/pgupgrade_controller.go

```diff
@@ -331,6 +344,22 @@ func (r *PGUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	setStatusToProgressingIfReasonWas("PGClusterNotShutdown", upgrade)
 
+	// A separate check for primary identification allows for cases where the
```

Comment:
What should a user do in this scenario?

Reply:
I'd like to investigate additional options, but, as of right now, I think the upgrade has to be abandoned.
This can result from when a PostgresCluster never initialized in the first place. In this case a new cluster should be created. The more complex scenario is when the cluster has previously initialized, but for some reason the correct StatefulSet cannot be identified.